### PR TITLE
Modernize the repo

### PR DESCRIPTION
**Pull request by mraarif**
`python-modernize` 
`isort`
`itertools.izip_longest`

**TODO items** (when ready to add support for `python3`)
need to replace `from cStringIO import StringIO` in tests. But right now tests are running fine on 3.5 even with this import. Debugging this issue.

### CODE REVIEW

**Comment by awais786** [comment]
@mraarif imports order still failing.

**Comment by awais786** [comment]
@jmbowman any idea why imports order quality failing ?

**Comment by jmbowman** [comment]
Add the `--diff` parameter to all `isort` calls in the `Makefile` to get more details.  Some of the earlier packages to use isort were configured before we realized this option existed.

File: .isort.cfg [hunk 1]
```diff
@@ -1,3 +1,8 @@
 [settings]
+default_section=FIRSTPARTY
```
Comment (awais786) [inline .isort.cfg:2]: imports order quality was failing on travis. Added these configs for sorting purpose.

**Comment by mraarif** [comment]
command `test-docker-py3` is failing (trying to execute tests against python 3)

**Comment by mraarif** [comment]
@jmbowman after @brianhw 's comment on #777 , do we still want to continue working on getting tests run for `python 3`?

**Comment by jmbowman** [comment]
This can wait until DE is ready to try moving forward with the Python 3 upgrade again.  The main outcome we wanted from [BOM-1041](https://openedx.atlassian.net/browse/BOM-1041) was to fix the code in sample-themes, edx-themes, and edx-microsite that were Python 3 incompatible.  It looks like that's done now, which should solve the problem as far as edx-platform is concerned.

File: edx/analytics/tasks/common/tests/test_mysql.py [hunk 2]
```diff
@@ -148,9 +148,9 @@ def test_connect_with_complete_credentials(self):
 
     def test_execute_query(self):
         self.mock_cursor.fetchone.side_effect = [
-            (2L,),
-            (3L,),
-            (10L,),
+            (2,),
```
Comment (awais786) [inline edx/analytics/tasks/common/tests/test_mysql.py:151]: why removing `L`

File: edx/analytics/tasks/insights/tests/test_module_engagement.py [hunk 3]
```diff
@@ -306,7 +308,7 @@ class ModuleEngagementSummaryDataTaskReducerTest(ReducerTestMixin, TestCase):
     input_record = ModuleEngagementRecord(
         course_id='foo/bar/baz',
         username='test_user',
-        date=datetime.date(2014, 03, 26),
+        date=datetime.date(2014, 0o3, 26),
```
Comment (awais786) [inline edx/analytics/tasks/insights/tests/test_module_engagement.py:311]: is it by `modernize` ?

**Review by awais786 — APPROVED** [review]

**Comment by awais786** [comment]
@jmbowman still no update on this repo ?

**Comment by jmbowman** [comment]
Hmm, I think we completely lost track of this when upgrading Insights and edx-analytics-data-api.  Yeah, we really should get this off of 2.7.  There's a good overview of the last attempt in https://github.com/edx/edx-analytics-pipeline/pull/776 , probably should start there and start ticketing the remaining parts.  Please coordinate with the eduNEXT team on this, since they did a lot of related work for the other analytics services.

File: edx/analytics/tasks/tests/acceptance/test_database_export.py [hunk 4]
```diff
@@ -56,7 +58,7 @@ def create_temporary_directories(self):
         for dir_path in [self.external_files_dir, self.working_dir, self.validation_dir, self.gpg_dir]:
             os.makedirs(dir_path)
 
-        os.chmod(self.gpg_dir, 0700)
+        os.chmod(self.gpg_dir, 0o700)
```
Comment (morenol) [inline edx/analytics/tasks/tests/acceptance/test_database_export.py:61]: Same that in the other comment

**Review by morenol — APPROVED** [review]

File: edx/analytics/tasks/tests/acceptance/test_database_export.py (hunk 4 above)
Comment (morenol) [reply edx/analytics/tasks/tests/acceptance/test_database_export.py:61]: Forget this, I just realized that that is the syntax for the numbers with leading zeros.

**Comment by morenol** [comment]
Hi @mraarif, could you add to the travis file a way to run this tests:
`make test-docker-py3` it is currently failing due to it is using an old version of ansible

 or should it be done in another PR? 

I think that we should have that test in a different travis worker

**Comment by jmbowman** [comment]
@morenol Is there a log of the failure somewhere?  Do we just need changes made to the image at https://hub.docker.com/r/edxops/analytics_pipeline/tags ?  That would seem like a configuration repo PR.

**Comment by morenol** [comment]
@jmbowman no, I ran that locally using the devstack, I think that we dont need to install anything else, just to run `make test-docker-py3` in the tests just like we run `make test-docker`

But in anyway I think that the changes of this PR are OK, and that the changes for the python3 tests should be done in another PR

**Comment by mraarif** [comment]
@jmbowman as noted by @morenol in this [PR](https://github.com/edx/edx-analytics-pipeline/pull/742#issuecomment-646067981), do we still have plans for this change to go ahead?

**Comment by jmbowman** [comment]
@mraarif As noted on the other PR, the DE team has suggested holding off on updating this repo because they plan to deprecate it and stop running the service in the relatively near future.

**Pull request closed without merging**